examples/firewall/firewall.c

```diff
@@ -0,0 +1,388 @@
+}
+
+static int
+packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta, __attribute__((unused)) struct onvm_nf_info *nf_info) {
+        struct ipv4_hdr *ipv4_hdr;
+        static uint32_t counter = 0;
+        uint32_t rule = 0;
+        uint32_t track_ip = 0;
+
+        if (onvm_pkt_is_ipv4(pkt)) {
+                ipv4_hdr = onvm_pkt_ipv4_hdr(pkt);
+                int ret = rte_lpm_lookup(lpm_tbl, ipv4_hdr->src_addr, &rule);
+
+                if (ret) {
+                        meta->action = ONVM_NF_ACTION_DROP;
+                        stats.pkt_drop++;
+                        stats.pkt_total++;
+                        if (debug) {
+                                RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n", ipv4_hdr->src_addr);
+                        }
+                } else {
+                        switch (rule) {
+                                case 0:
+                                        meta->action = ONVM_NF_ACTION_TONF;
+                                        meta->destination = destination;
+                                        stats.pkt_accept++;
+                                        stats.pkt_total++;
+                                        if (debug) {
+                                                RTE_LOG(INFO, APP, "Packet from source IP %d has been accepted.\n",
+                                                        ipv4_hdr->src_addr);
+                                        }
+                                        break;
+                                default:
+                                        meta->action = ONVM_NF_ACTION_DROP;
+                                        stats.pkt_drop++;
+                                        stats.pkt_total++;
+                                        if (debug) {
+                                                RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n",
+                                                        ipv4_hdr->src_addr);
+                                        }
+                                        break;
+                        }
+                }
+        } else {
+                if (debug) {
+                        RTE_LOG(INFO, APP, "Packet received not ipv4\n");
+                }
+                stats.pkt_drop++;
+                stats.pkt_total++;
+                meta->action = ONVM_NF_ACTION_DROP;
+        }
+
+        if (++counter == print_delay) {
+                do_stats_display();
+                counter = 0;
+        }
+
+        return 0;
+}
+
+static int lpm_setup(struct onvm_fw_rule **rules, int num_rules) {
+        int i, status;
+
+        firewall_req = (struct lpm_request *) rte_malloc(NULL, sizeof(struct lpm_request), 0);
+
+        if (!firewall_req) return 0;
+
+        firewall_req->max_num_rules = 1024;
+        firewall_req->num_tbl8s = 24;
+        firewall_req->socket_id = rte_socket_id();
+        firewall_req->name[0] = 'f';
```

Comment:
Lets make the name unique,see this example https://github.com/sdnfv/openNetVM-dev/blob/7ae6d81c148d43af39d81127bdce8d0d0b4e2690/onvm/onvm_nflib/onvm_flow_table.c#L78